### PR TITLE
dev-python/flask-paginate: bumped to 0.7.0

### PR DESCRIPTION
**Pull request by florianjr**
Signed-off-by: Florian Förster <foerster.florianjr@gmail.com>
Closes: https://bugs.gentoo.org/717526
Package-Manager: Portage-3.0.1, Repoman-2.3.23

I copied the existing ebuild and changed the $MY_COMMIT variable. @stkw0 i removed you from the maintainers list and added myself, as we said earlier in the previous PR.

### CODE REVIEW

**Comment by florianjr** [comment]
Are there any more steps to become the maintainer than adding myself to the metadata.xml? Am i then automatically a proxy-maintainer?

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+PYTHON_COMPAT=( pypy3 python3_{6,7} )
+
+inherit distutils-r1
+
+MY_COMMIT="6445a67395912a5432014ec824ce4bf8fd13e22d"
+
+DESCRIPTION="Pagination support for flask"
+HOMEPAGE="https://flask-paginate.readthedocs.io"
+# https://github.com/lixxu/flask-paginate/issues/68
+SRC_URI="https://github.com/lixxu/flask-paginate/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
```
Comment (stkw0) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:15]: it should be `SRC_URI="https://github.com/lixxu/flask-paginate/archive/${PV}.tar.gz"` as we said in the previous PR.

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+PYTHON_COMPAT=( pypy3 python3_{6,7} )
+
+inherit distutils-r1
+
+MY_COMMIT="6445a67395912a5432014ec824ce4bf8fd13e22d"
+
+DESCRIPTION="Pagination support for flask"
+HOMEPAGE="https://flask-paginate.readthedocs.io"
+# https://github.com/lixxu/flask-paginate/issues/68
```
Comment (stkw0) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:14]: remove this line as this issue is no longer relevant

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild (hunk 1 above)
Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:15]: I tried it, the $S was no more working, also when I deleted it (and used the default value) put the PV in there. I mean it’s working.

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:14]: In dev-python/flask-themes it's also included for example.

Comment (stkw0) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:14]: I didn't understood that.

I tried this and it seems it's working
```diff
 inherit distutils-r1

-MY_COMMIT="510ad833106134711868653fb597bf75ea8ac34f"
-
 DESCRIPTION="Pagination support for flask"
 HOMEPAGE="https://flask-paginate.readthedocs.io"
-# https://github.com/lixxu/flask-paginate/issues/68
-SRC_URI="https://github.com/lixxu/flask-paginate/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+SRC_URI="https://github.com/lixxu/flask-paginate/archive/${PV}.tar.gz"

 LICENSE="BSD"
 SLOT="0"
@@ -26,8 +23,6 @@ BDEPEND="
        test? ( dev-python/pytest[${PYTHON_USEDEP}] )
 "

-S="${WORKDIR}/${PN}-${MY_COMMIT}"
-
 python_test() {
        pytest -vv tests/tests.py || die "tests failed with ${EPYTHON}"
 }
```

btw there can be more than one maintainer. But I'm not specially interested on this pacakge so I don't have any problem to be removed from the maintainer list. did you ask in #gentoo-proxy-maint IRC channel? I think a dev have to open a ticket for you as proxy maintainer

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:14]: Sorry for my English here :) It doesn't work on my machine. I get a "The source directory '/var/tmp/portage/dev-python/flask-paginate-0.7.0/work/flask-paginate-0.7.0' doesn't exist". Should i just create this directory? I can't send messages in #gentoo-proxy-maint, it's a "can't send nick/channel". I tried several clients including Kiwi Web.

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:14]: Alright i forgot to remove the -> ${P}.tar.gz

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild]: Now its working :)

Comment (stkw0) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild]: nice. I forgot that tarball should contain package name. 

You should fixup all commits and leave only one commit.

About the IRC channel, maybe you need to register your user.

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( pypy3 python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Pagination support for flask"
+HOMEPAGE="https://flask-paginate.readthedocs.io"
+SRC_URI="https://github.com/lixxu/flask-paginate/archive/${PV}.tar.gz -> ${P}-${PV}.tar.gz"
```
Comment (stkw0) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:12]: ${P} variable in this case is the same as ${PN}-${PV}. Please leave it as ${P}.tar.gz

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:12]: I know and understand this, but could you please explain to me, why i'm then getting: "The source directory ${S} doesn't exist" ?

Comment (florianjr) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:12]: Should we define a $S ?

Comment (stkw0) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:12]: I'm not getting those errors. Check which source directory is defined and the directory name on PORTAGE_TMPDIR

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+RDEPEND="dev-python/flask[${PYTHON_USEDEP}]"
+BDEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( dev-python/pytest[${PYTHON_USEDEP}] )
+"
+
+python_test() {
+	pytest -vv tests/tests.py || die "tests failed with ${EPYTHON}"
+}
```
Comment (juippis) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:28]: You can replace all of these lines with a single `distutils_enable_tests pytest`

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+DESCRIPTION="Pagination support for flask"
+HOMEPAGE="https://flask-paginate.readthedocs.io"
+SRC_URI="https://github.com/lixxu/flask-paginate/archive/${PV}.tar.gz -> ${P}-${PV}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
```
Comment (juippis) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:18]: These will be handled by `distutils_enable_tests pytest` as well, but put it under `RDEPEND`.

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild (hunk 3 above)
Comment (juippis) [reply dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:12]: Tarball name has nothing to do with `S` structure. You can inspect the tarball by simply opening/extracting it, and/or browse to `PORTAGE_TMPDIR` as suggested when emerge fails.

File: dev-python/flask-paginate/flask-paginate-0.7.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( pypy3 python3_{6,7} )
```
Comment (juippis) [inline dev-python/flask-paginate/flask-paginate-0.7.0.ebuild:6]: Can you add 3.8 and 3.9 support here? Seems like they should work.

**Pull request closed without merging**